web/satellite/src/components/account/billing/paymentMethods/AddTokenCard.vue

```diff
@@ -0,0 +1,227 @@
+} = PAYMENTS_ACTIONS;
+
+// @vue/component
+@Component({
+    components: {
+        StorjLarge,
+        VButton,
+        TokenDepositSelection2,
+    },
+})
+export default class AddTokenCard extends Vue {
+    @Prop({default: 0})
+    private readonly totalCount: number;
+    @Prop({default: false})
+    private readonly DEFAULT_TOKEN_DEPOSIT_VALUE = 10; // in dollars.
+    private readonly MAX_TOKEN_AMOUNT = 1000000; // in dollars.
+    private tokenDepositValue: number = this.DEFAULT_TOKEN_DEPOSIT_VALUE;
+
+    public toggleShowAddFunds(): void {
+        this.$emit("showAddFunds")
+    }
+
+    /**
+     * Set of default payment options.
+     */
+    public paymentOptions: PaymentAmountOption[] = [
+        new PaymentAmountOption(10, `USD $10`),
+        new PaymentAmountOption(20, `USD $20`),
+        new PaymentAmountOption(50, `USD $50`),
+        new PaymentAmountOption(100, `USD $100`),
+        new PaymentAmountOption(1000, `USD $1000`),
+    ];
+
+    /**
+     * onConfirmAddSTORJ checks if amount is valid.
+     * If so processes token payment and returns state to default.
+     */
+    public async onConfirmAddSTORJ(): Promise<void> {
+
+        if (!this.isDepositValueValid) return;
+
+        try {
+            const tokenResponse = await this.$store.dispatch(MAKE_TOKEN_DEPOSIT, this.tokenDepositValue * 100);
+            await this.$notify.success(`Successfully created new deposit transaction! \nAddress:${tokenResponse.address} \nAmount:${tokenResponse.amount}`);
+            const depositWindow = window.open(tokenResponse.link, '_blank');
+            if (depositWindow) {
+                depositWindow.focus();
+            }
+            location.reload()
+            this.toggleShowAddFunds();
+        } catch (error) {
+            await this.$notify.error(error.message);
+            this.$emit('toggleIsLoading');
+        }
+
+        this.tokenDepositValue = this.DEFAULT_TOKEN_DEPOSIT_VALUE;
+        try {
+            await this.$store.dispatch(GET_PAYMENTS_HISTORY);
+        } catch (error) {
+            await this.$notify.error(error.message);
+            this.$emit('toggleIsLoading');
+        }
+    }
+
+    /**
+     * Event for changing token deposit value.
+     */
+    public onChangeTokenValue(value: number): void {
+        this.tokenDepositValue = value;
+    }
+
+    /**
+     * Indicates if deposit value is valid.
+     */
+    private get isDepositValueValid(): boolean {
+        switch (true) {
+        case (this.tokenDepositValue < this.DEFAULT_TOKEN_DEPOSIT_VALUE || this.tokenDepositValue >= this.MAX_TOKEN_AMOUNT) && !this.userHasOwnProject:
+            this.$notify.error('First deposit amount must be more than $10 and less than $1000000');
+            this.setDefault();
+
+            return false;
+        case this.tokenDepositValue >= this.MAX_TOKEN_AMOUNT || this.tokenDepositValue === 0:
+            this.$notify.error('Deposit amount must be more than $0 and less than $1000000');
```

Comment:
nit - I would prefer to use the variable in the message here, `MAX_TOKEN_AMOUNT` rather than hardcoding the amount here